app/src/main/java/ltd/evilcorp/atox/activity/SettingsActivity.kt

```diff
@@ -0,0 +1,147 @@
+package ltd.evilcorp.atox.activity
+
+import android.content.SharedPreferences
+import android.os.Bundle
+import android.util.Log
+import android.view.MenuItem
+import androidx.appcompat.app.AppCompatActivity
+import androidx.preference.Preference
+import androidx.preference.PreferenceFragmentCompat
+import androidx.preference.PreferenceManager
+import ltd.evilcorp.atox.App
+import ltd.evilcorp.atox.R
+import ltd.evilcorp.atox.tox.ToxThread
+
+private const val TITLE_TAG = "settingsActivityTitle"
+
+class SettingsActivity : AppCompatActivity(),
+    PreferenceFragmentCompat.OnPreferenceStartFragmentCallback,
+    SharedPreferences.OnSharedPreferenceChangeListener {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.settings_activity)
+        if (savedInstanceState == null) {
+            supportFragmentManager
+                .beginTransaction()
+                .replace(R.id.settings, HeaderFragment())
+                .commit()
+        } else {
+            title = savedInstanceState.getCharSequence(TITLE_TAG)
+        }
+        supportFragmentManager.addOnBackStackChangedListener {
+            if (supportFragmentManager.backStackEntryCount == 0) {
+                setTitle(R.string.title_activity_settings)
+            }
+        }
+        supportActionBar?.setDisplayHomeAsUpEnabled(true)
+    }
+
+    override fun onSharedPreferenceChanged(prefs: SharedPreferences?, key: String?) {
+        Log.e("Settings", "Key changed $key")
+        when {
+            key.equals("username") -> {
+                with(App.toxThread.handler) {
+                    sendMessage(
+                        obtainMessage(
+                            ToxThread.msgSetName,
+                            0,
+                            0,
+                            prefs!!.getString(key, getString(R.string.name_default))
+                        )
+                    )
+                }
+            }
+            key.equals("status_message") -> {
+                with(App.toxThread.handler) {
+                    sendMessage(
+                        obtainMessage(
+                            ToxThread.msgSetStatus,
+                            0,
+                            0,
+                            prefs!!.getString(key, getString(R.string.status_message_default))
+                        )
+                    )
+                }
+            }
+            key.equals("status") -> {
+                with(App.toxThread.handler) {
+                    sendMessage(obtainMessage(ToxThread.msgSetState, prefs!!.getInt(key, 0), 0, null))
+                }
+            }
+        }
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem?): Boolean {
+        var id = item!!.itemId
```

Comment:
Remove this variable after fixing the MenuItem? thing. (I get a lint warning on this line about that this should be val instead of var, but we don't really need it if MenuItem isn't nullable.)